src/Avalonia.Controls/ToolTip.cs

```diff
@@ -80,6 +81,28 @@ public class ToolTip : ContentControl, IPopupHostProvider
         internal static readonly AttachedProperty<ToolTip?> ToolTipProperty =
             AvaloniaProperty.RegisterAttached<ToolTip, Control, ToolTip?>("ToolTip");
 
+        /// <summary>
+        /// The event raised when a ToolTip is going to be shown on an element.
+        /// </summary>
+        /// <remarks>
+        /// To prevent a tooltip from appearing in the UI, your handler for ToolTipOpening can mark the event data handled.
+        /// Otherwise, the tooltip is displayed, using the value of the ToolTip property as the tooltip content.
+        /// Another possible scenario is that you could write a handler that resets the value of the ToolTip property for the element that is the event source, just before the tooltip is displayed.
+        /// ToolTipOpening will not be raised if the value of ToolTip is null or otherwise unset. Do not deliberately set ToolTip to null while a tooltip is open or opening; this will not have the effect of closing the tooltip, and will instead create an undesirable visual artifact in the UI.
+        /// </remarks>
+        public static readonly RoutedEvent ToolTipOpeningEvent =
+            RoutedEvent.Register<ToolTip, RoutedEventArgs>("ToolTipOpening", RoutingStrategies.Direct);
```

Comment:
As discussed new events here should follow existing conventions and use `CancelRoutedEventArgs`. This follows Expander which itself was based on what Flyout did (although flyout isn't API symmetric and doesn't allow cancelling an open).